axelor-account/src/main/java/com/axelor/apps/account/db/repo/AccountInvoiceTermRepository.java

```diff
@@ -0,0 +1,19 @@
+package com.axelor.apps.account.db.repo;
+
+import com.axelor.apps.account.db.InvoiceTerm;
+import com.axelor.apps.account.db.PaymentSession;
+import com.axelor.apps.base.db.Partner;
+import com.axelor.db.Query;
+import java.util.List;
+
+public class AccountInvoiceTermRepository extends InvoiceTermRepository {
+
+  public List<InvoiceTerm> findByPaymentSessionAndPartner(
+      PaymentSession paymentSession, Partner partner) {
+    return Query.of(InvoiceTerm.class)
+        .filter("self.paymentSession.id = :id AND self.invoice.partner.id = :partnerId")
+        .bind("id", paymentSession.getId())
+        .bind("partnerId", partner)
```

Comment:
I think it works because of a AOP hack, but it's cleaner to give partner id to the query here to avoid issue in case of future change.